memory-viz/src/memory_model.ts

```diff
@@ -1145,4 +1204,96 @@ export class MemoryModel {
             this.svg.setAttribute("height", this.height.toString());
         }
     }
+
+    /**
+     * Add hover interactivity to the SVG on object IDs
+     */
+    setInteractivityScript(): void {
+        const idToObjectMapping = Object.fromEntries(this.idToObjectMap);
+
+        const script = `
+            function objectInteractivity() {
+                // Inject the id value to object id mapping into script
+                const idToObjectMap = ${JSON.stringify(idToObjectMapping)};
+                
+                function highlightObject(objectId) {
+                    const objectBox = document.getElementById(objectId);
+                    if (!objectBox) return;
+                    
+                    const bbox = objectBox.getBBox();
+                    const existingFill = objectBox.querySelector('path[fill]:not([fill="none"])');
+                    
+                    if (existingFill) {
+                        existingFill.setAttribute('data-original-fill', existingFill.getAttribute('fill'));
+                        existingFill.setAttribute('fill', 'rgba(255, 255, 0, 0.6)');
+                    } else {
+                        const highlight = document.createElementNS("http://www.w3.org/2000/svg", "rect");
```

Comment:
Adding dynamic rectangles is not a bad idea, but a bit clunky. (Also I think the current implementation will keep on adding new `rect` elements to the same object each time this function is called.

Looking at the roughjs API, I think if you modify the calls to `roughjs.rectangle` to _always_ have a `fillStyle` of `'solid'` (and by default a `fill` value of `'none'`) you can ensure a closed path is always drawn, which will allow you to set the `fill` of that path rather than needing to create a new `rect`.